src/providers/abstract-provider.ts

```diff
@@ -307,12 +314,17 @@ async function getSubscription(_event: ProviderEvent, zone?: Zone): Promise<Subs
             case 'safe': {
                 return { type: _event, tag: _event };
             }
+            case 'qiTransaction': {
+                assert(zone != null, 'zone is required for qiTransaction event', 'MISSING_ARGUMENT');
+                assert(hash != null, 'hash is required for qiTransaction event', 'MISSING_ARGUMENT');
+                return { type: _event, tag: _event, hash, zone };
+            }
         }
     }
 
     if (isHexString(_event, 32)) {
         const hash = _event.toLowerCase();
-        zone = toZone(hash.slice(0, 4));
+        zone = zone ?? toZone(hash.slice(0, 4));
         return { type: 'transaction', tag: getTag('tx', { hash }), hash, zone };
     }
```

Comment:
Why did we add an extra hash argument to the getSubscription method if we are already handling hash events for regular transactions? Cant we just use that same pattern?